qiskit_ibm_runtime/accounts/account.py

```diff
@@ -48,17 +64,53 @@ def _assert_valid_url(url: str) -> None:
     try:
         urlparse(url)
     except:
-        raise ValueError(f"Inappropriate `url` value. Failed to parse '{url}' as URL.")
+        raise ValueError(f"Invalid `url` value. Failed to parse '{url}' as URL.")
+
+
+def _assert_valid_proxies(config: ProxyConfigurationType) -> None:
+    """Assert that the proxy configuration is valid."""
+
+    if config is None:
+        return
+
+    # verify NTLM authentication configuration
+    ntlm_user = config.get("username_ntlm")
+    ntlm_pass = config.get("password_ntlm")
+    if not any(
+        [
+            isinstance(ntlm_user, str) and isinstance(ntlm_pass, str),
+            ntlm_user is None and ntlm_pass is None,
+        ]
+    ):
+        raise ValueError(
+            f"Invalid proxy configuration for NTLM authentication. None or both of username and "
+            f"password must be provided. Got username_ntlm={ntlm_user}, password_ntlm={ntlm_pass}."
+        )
+
+    # verify proxy configuration
+    urls = config.get("urls")
+    if urls is not None and not isinstance(urls, dict):
+        raise ValueError(
+            f"Invalid proxy configuration. Expected `urls` to contain a dictionary mapping protocol "
+            f"or protocol and host to the URL of the proxy. Got {urls}"
+        )
 
 
 def _assert_valid_instance(auth: AccountType, instance: str) -> None:
     """Assert that the instance name is valid for the given account type."""
     if auth == "cloud":
         if not (isinstance(instance, str) and len(instance) > 0):
             raise ValueError(
-                f"Inappropriate `instance` value. Expected a non-empty string."
+                f"Invalid `instance` value. Expected a non-empty string, got '{instance}'."
             )
-    # TODO: add validation for legacy instance when adding test coverage
+    if auth == "legacy":
+        if instance is not None:
+            try:
+                from_instance_format(instance)
+            except:
+                raise ValueError(
+                    f"Invalid `instance` value. Expected hub/group/project format, got {instance}"
+                )
```

Comment:
Required updates in other test cases that passed in invalid values indirectly, see https://github.com/Qiskit/qiskit-ibm-runtime/pull/80/commits/90942c78d8a7164e96035dba03e58a8d9eb48b24